### DISCUSSION TITLE
What kind of protocol it posible to use for loading file\image

### DISCUSSION THREAD

**Discussion by mabps** [ipxe/ipxe#638] [General]
Sorry if my question repeats, but I couldn't find the FAQ answer for the last 2 years
What are the ways to transfer a menu file to ipxe.efi? using which protocol - http, ftp, tftp?
In the example documentation on your site - you pass the menu file "real_boot_script.php " according to the protocol of HTTP. is it possible to prescribe the use of another protocol and how to do it ?

example:
```
option client-architecture code 93 = unsigned integer 16;
if exists user-class and option user-class = "iPXE" {
       filename "http://my.web.server/real_boot_script.php";
 } else if option client-architecture = 00:00 {
       filename "undionly.kpxe";
} else {
       filename "ipxe.efi";
}
```

can i wrote line "`filename "http://my.web.server/real_boot_script.php";`"  as "`filename "maymenu.ipxe";`" 
Will the iPXE loader download this file over the tftp protocol ? or it's error ?

**Comment by stappersg** [comment]
The `"maymenu.ipxe"` has no protocol, fall back to _TFTP_.

A `"http://servername/path/maymenu"` does specify _HTTP_ and _server name_.


About the
> Will the iPXE loader download this file over the tftp protocol ? or it's error ?

[Just try it and learn something new](http://www.catb.org/~esr/faqs/smart-questions.html)

**Reply by stappersg** [reply]
Oops, I meant

A `"http://servername/path/maymenu.ipxe"` does specify _HTTP_ and _server name_.

But no worries about an _"extension"_ in the HTTP request.
If I recall correct, is even the `Content-Type: ` not important.  Play safe and go for  `text/plain`.
Important is that the first line in the text file is `#!ipxe`.

**Reply by mabps** [reply]
thanks for you answer
apparently we will have to try several options

**Comment by NiKiZe** [comment]
Depending on which protocols is enabled during compilation you can use URLs such as "tftp://server/file" or "nfs://server/file" http is the most common.

iPXE supports ipxe scripts, you can ginde the details about scripting on ipxe.org, one feature of the scripting language is menus and some other logic.

**Reply by mabps** [reply]
thanks for you anser
I have already read about scripts for the site ipxe.org and in all the examples, only the http protocol is used there, it is permissible to use other protocols it is not written there, or I read and did not understand :-)
